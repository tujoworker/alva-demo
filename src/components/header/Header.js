/**
 * Component
 *
 * @author Tobias HÃ¸egh <tobias@tujo.no>
 */

import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'

import styled from 'react-emotion'

import ChangeLang from './ChangeLang'
import Menu from './Menu'

const HeaderComp = styled('header')`
  ${'' /* grid-column-start: main-start;
  grid-column-end: main-end;
  grid-row-start: main-start; */} height: 10em;
  /*margin-top: 1.5em;*/

  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;

  font-size: 1.3em; /* We could use rem; here if HTML element has synamic font sizing! */
  background: blue;
`

// console.log('styles', style)

// @withRouter
// // @injectIntl
// @inject('store')
// @observer
export default class Header extends Component {
  // docs (or use ptd): https://github.com/facebook/prop-types#usage
  static propTypes = {
    foo: PropTypes.string
  }
  static defaultProps = {
    foo: ''
  }
  // componentDidMount() {
  // 	anime({
  // 	    targets: 'header',
  // 	    easing: 'easeOutExpo',
  // 	    // translateY: [{ value: -10, duration: 0 }, 0],
  // 	    backgroundColor: ['rgb(255, 50, 50)', 'rgb(255, 0, 0)'],
  // 	    duration: 2e3
  // 	})
  // }

  render() {
    // const { formatMessage } = this.props.intl
    // let { locale } = this.props
    // const c = this.props.locale.L('hello')
    // const c = this.props.locale
    // const c = this.props.locale.formatMessage('hello')
    // console.log('locale 1', c)
    // const c2 = this.props.locale.formatMessage({
    //     id: 'hello'
    //     // defaultMessage: 'X',
    // })
    // console.log('locale 2', c2)
    // const c3 = this.props.intl.formatMessage({
    //     id: 'hello'
    //     // defaultMessage: 'X',
    // })
    // console.log('locale 3', c3)

    return (
      <HeaderComp role="banner">
        <Menu />
        <ChangeLang />
        <Fragment>
          <span>Fragment!</span>
        </Fragment>
        {/* <FormattedMessage id="hello" values={{ name: <b>{'x'}</b> }} /> */}
        {/* {this.props.intl.formatMessage(
                    {
                        id: 'hello'
                        // defaultMessage: 'X',
                    },
                    { name: 'my name' }
                )} */}
        {/* <span>{this.props.locale.L('hello', { name: 'x' })}</span> */}
        {/* <span>{this.props.store.email}</span> */}
        {/* <p>
                    <FormattedNumber
                        value={1000}
                        style="currency"
                        currency="NOK"
                    />
                </p> */}
        {/* <Anime
                    easing="easeOutElastic"
                    duration={1e3}
                    scale={[0.9, 1]}
                    opacity={[0, 1]}
                    delay={1e3}
					>
                </Anime> */}
      </HeaderComp>
    )
  }
}
