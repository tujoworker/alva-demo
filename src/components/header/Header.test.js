/**
 * A simple test, to test the results of components
 *
 * @author Tobias HÃ¸egh <tobias@tujo.no>
 */

//test core packages
import { shallow, fakeProps } from 'alva/tdd'
import React from 'react'

// import fakeProps2 from 'react-fake-props'
// import path from 'path'
import toJson from 'enzyme-to-json'

import Header from './Header'

// The API Docs
// http://airbnb.io/enzyme/docs/api/
// https://facebook.github.io/jest/docs/en/expect.html

const store = fakeProps(require.resolve('./Header'), { optional: true })
// const store = {}
// console.log('store', store)

describe('Component renders correctly', () => {
  const HeaderWrap = shallow(<Header {...store} />)

  it('have to match snapshot', () => {
    expect(toJson(HeaderWrap)).toMatchSnapshot()
  })

  it('has a role attribute', () => {
    expect(HeaderWrap.find('[role="banner"]').exists()).toBe(true)
    expect(HeaderWrap.first().props().role).toBe('banner')
  })

  it('has a menu inside', () => {
    expect(HeaderWrap.closest('nav').exists()).toBe(true)
  })

  // it('has to have at least one li element', () => {
  //     expect(HeaderWrap.find('li.list-group-item')).toHaveLength(1);
  // });
})
