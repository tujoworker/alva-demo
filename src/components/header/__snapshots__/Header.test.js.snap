// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly have to match snapshot 1`] = `
<Styled(header)
  role="banner"
>
  <inject-Menu-with-locale />
  <inject-MyComponent-with-locale />
  <React.Fragment>
    <span>
      Fragment!
    </span>
  </React.Fragment>
</Styled(header)>
`;
